solidity/script/avs/DeployAVS.s.sol

```diff
@@ -0,0 +1,134 @@
+import {TransparentUpgradeableProxy} from "../../contracts/upgrade/TransparentUpgradeableProxy.sol";
+import {ECDSAStakeRegistry} from "../../contracts/avs/ECDSAStakeRegistry.sol";
+import {Quorum, StrategyParams} from "../../contracts/interfaces/avs/vendored/IECDSAStakeRegistryEventsAndErrors.sol";
+import {HyperlaneServiceManager} from "../../contracts/avs/HyperlaneServiceManager.sol";
+
+import {TestPaymentCoordinator} from "../../contracts/test/avs/TestPaymentCoordinator.sol";
+
+contract DeployAVS is Script {
+    using stdJson for string;
+
+    struct StrategyInfo {
+        string name;
+        address strategy;
+    }
+
+    uint256 deployerPrivateKey;
+
+    ProxyAdmin public proxyAdmin;
+    IAVSDirectory public avsDirectory;
+    IPaymentCoordinator public paymentCoordinator;
+    IDelegationManager public delegationManager;
+
+    Quorum quorum;
+    uint256 thresholdWeight = 6667;
+
+    function _loadEigenlayerAddresses(string memory targetEnv) internal {
+        string memory root = vm.projectRoot();
+        string memory path = string.concat(
+            root,
+            "/script/avs/eigenlayer_addresses.json"
+        );
+        string memory json = vm.readFile(path);
+
+        proxyAdmin = ProxyAdmin(
+            json.readAddress(
+                string(abi.encodePacked(".", targetEnv, ".proxyAdmin"))
+            )
+        );
+        avsDirectory = IAVSDirectory(
+            json.readAddress(
+                string(abi.encodePacked(".", targetEnv, ".avsDirectory"))
+            )
+        );
+        delegationManager = IDelegationManager(
+            json.readAddress(
+                string(abi.encodePacked(".", targetEnv, ".delegationManager"))
+            )
+        );
+        // paymentCoordinator = IPaymentCoordinator(json.readAddress(string(abi.encodePacked(".", targetEnv, ".paymentCoordinator"))));
+        paymentCoordinator = new TestPaymentCoordinator(); // temporary until Eigenlayer deploys the real one
+
+        StrategyInfo[] memory strategies = abi.decode(
+            vm.parseJson(
+                json,
+                string(abi.encodePacked(".", targetEnv, ".strategies"))
+            ),
+            (StrategyInfo[])
+        );
+
+        StrategyParams memory strategyParam;
+
+        uint96 totalMultipliers = 10_000;
+        uint96 multiplier;
+
+        uint96 strategyCount = uint96(strategies.length);
+        for (uint96 i = 0; i < strategyCount; i++) {
+            // the multipliers need to add up to 10,000, so we divide the total by the number of strategies for the first n-1 strategies
+            // and then the last strategy gets the remainder
+            if (i < strategies.length - 1) {
+                multiplier = totalMultipliers / uint96(strategyCount);
+            } else {
+                multiplier =
+                    totalMultipliers -
+                    multiplier *
+                    uint96(strategyCount - 1);
```

Comment:
i dont understand this tbh

Reply:
I'll give you an example, 

say strategyCount = 3, then, then the strategy weights are 3333, 3333, 3334 to add upto to 10000.